python/pyiceberg/io/pyarrow.py

```diff
@@ -505,7 +505,9 @@ def project_table(
         # Get the schema
         with fs.open_input_file(path) as fout:
             parquet_schema = pq.read_schema(fout)
-            schema_raw = parquet_schema.metadata.get(ICEBERG_SCHEMA)
+            schema_raw = None
+            if parquet_schema.metadata is not None:
```

Comment:
You could also leverage the Walrus operator here:
```suggestion
            if metadata := parquet_schema.metadata:
```

Reply:
Oh sweet, I never knew about this operator, that's pretty nifty :) Thanks @Fokko !